bin/oneoff/census/state-data/upload-state-data.rb

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../../dashboard/config/environment'
+require 'optparse'
+
+CDO.log = Logger.new(STDERR)
+
+options = {
+  filename: nil,
+  state: nil,
+  school_year: nil,
+}
+OptionParser.new do |opts|
+  opts.banner = "Usage: #{File.basename(__FILE__)} [options]"
+  opts.on('-f', '--filename FILENAME', 'File containing the data to upload') do |f|
+    options[:filename] = f
+  end
+  opts.on('-s', '--state STATE_ABBREVIATION', 'Two letter state abbreviation') do |s|
+    options[:state] = s
+  end
+  opts.on('-y', '--school_year YEAR', 'The school year for the data. This should be the calendar year in which the school year began (e.g., 2017 for the 2017-2018 school year)') do |y|
+    options[:school_year] = y
+  end
+  opts.on('-h', '--help', 'Prints this help') do
+    puts opts
+    exit
+  end
+end.parse!
+
+missing_args = [:filename, :state, :school_year].select {|arg| options[arg].nil?}
+raise OptionParser::MissingArgument.new(missing_args.join(', ')) unless missing_args.empty?
+raise OptionParser::InvalidOption.new("State code must be two characters") unless options[:state].length == 2
```

Comment:
Is it worth validating against the actual state code [list](https://github.com/code-dot-org/code-dot-org/blob/a36c1ddcb536750558c640330e13395555eb1924/lib/state_abbr.rb#L42) (`STATE_ABBR_HASH.keys`)?

Reply:
Change this to check if the state code is valid.